### PR TITLE
build: migrate CRA to Vite

### PR DESCRIPTION
**Pull request by colbr**
### What does this do?

- Migrate from CRACO/CRA to Vite for all builds and local development
- Configures Vitest to work alongside Jest. This is so we can retain old tests while writing new ones. `*.vitest.*` will be handled by Vitest, and `*.test.*` will be handled by Jest.
- Configures Vitest and `testing-library/react` to work with each other.
- Adds tests for `AdminMessage` container as a working example of Vitest + `testing-library/react` + Redux.
- Switches `number.test.ts` to `number.vitest.ts` as a working example of Vitest.
- Sets up a Github Action to run Vite tests in parallel with Jest tests.

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -31,11 +31,11 @@ jobs:
               await github.rest.repos.createRelease({
                 draft: true,
                 generate_release_notes: true,
-                name: process.env.RELEASE_TAG,
+                name: import.meta.env.RELEASE_TAG,
                 owner: context.repo.owner,
                 prerelease: false,
                 repo: context.repo.repo,
-                tag_name: process.env.RELEASE_TAG,
+                tag_name: import.meta.env.RELEASE_TAG,
```
Comment (ratik21) [inline .github/workflows/create-release.yml:38]: to confirm VITE requires you to use `import.meta` ?

**Review by ratik21 — CHANGES_REQUESTED** [review]

File: .env.example [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
+VITE_VIDEO_ASSETS_PATH="https://res.cloudinary.com/fact0ry/video/upload/v1681745540/zero-assets/zos/"
 
-REACT_APP_IMAGE_ASSETS_PATH="https://res.cloudinary.com/fact0ry/image/upload/v1681745540/zero-assets/zos"
+VITE_IMAGE_ASSETS_PATH="https://res.cloudinary.com/fact0ry/image/upload/v1681745540/zero-assets/zos"
 
-REACT_APP_ANDROID_STORE_PATH="https://play.google.com/store/apps/details?id=com.zero.android"
+VITE_ANDROID_STORE_PATH="https://play.google.com/store/apps/details?id=com.zero.android"
 
-REACT_APP_MATRIX_HOME_SERVER_URL='https://zero-synapse-development-db365bf96189.herokuapp.com'
+VITE_MATRIX_HOME_SERVER_URL='https://zero-synapse-development-db365bf96189.herokuapp.com'
```
[the remark this answers is not in the document]
Comment (colbr) [reply .env.example:25]: [By default client-side variables in Vite need to be prepended with `VITE_`](https://vitejs.dev/guide/env-and-mode#env-files), so it's a requirement. But, turns out you can actually customise the variable prefix, so I could change it to `REACT_APP_` if that's preferable?

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (colbr) [reply .github/workflows/create-release.yml:38]: Yep yep, [`import.meta` is how it's done in Vite](https://vitejs.dev/guide/env-and-mode#env-files). There might be a way to cast it to `process` instead if need be.

File: src/authentication/create-account-method/styles.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@use '~@zero-tech/zui/styles/theme' as theme;
+@use '@zero-tech/zui/styles/theme' as theme;
```
[the remark this answers is not in the document]
Comment (colbr) [reply src/authentication/create-account-method/styles.scss:1]: Yep, `~` isn't required in Vite. I could probably adjust the config such that we don't have to change the `~` if you'd prefer that?

File: .env.example (hunk 2 above)
[the remark this answers is not in the document]
Comment (domw30) [reply .env.example:25]: Don't think this really matters too much but if its easy to customise the prefix I'd go with Ratiks suggestion just to minimise changes

**Review by domw30 — APPROVED** [review]
LFG

File: .env.example (hunk 2 above)
[the remark this answers is not in the document]
Comment (colbr) [reply .env.example:25]: Roger - I'll switch the prefix then change env in a follow up PR 🤝

File: src/authentication/create-account-method/styles.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (ratik21) [reply src/authentication/create-account-method/styles.scss:1]: got it. nah was just curious, thanks for clarifying

File: .env.example (hunk 2 above)
[the remark this answers is not in the document]
Comment (colbr) [reply .env.example]: Prefix is changed 🦜

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (colbr) [reply .github/workflows/create-release.yml]: I cast it to `process.env` so we could change less of the code.

**Review by domw30 — COMMENTED** [review]
Added notes to linear @colbr  💪

File: src/components/chat-view-container/chat-view.test.tsx [hunk 4]
```diff
@@ -286,13 +286,14 @@ describe('ChatView', () => {
     });
 
     it('returns the formatted date for the same year', () => {
-      const currentYearDate = moment('2024-06-11'); // Example date within the same year
+      const currentYearDate = moment('2024-05-11'); // Example date within the same year
```
[the remark this answers is not in the document]
Comment (colbr) [reply src/components/chat-view-container/chat-view.test.tsx:289]: Haha yes, that was extremely confusing!

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (ratik21) [reply .github/workflows/create-release.yml]: yeah, also just for note, this was a github workflow file, so it won't be included in the build.

File: src/components/admin-message/container.vitest.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            },
+          },
+          normalized: {
+            users: {
+              'inviter-id': { id: 'inviter-id', firstName: 'Courtney' },
+            },
+          },
+        },
+      }
+    );
```
Comment (ratik21) [inline src/components/admin-message/container.vitest.tsx:33]: I think we may need to iterate on this a bit. I like the way how we currently do it (using `subject({ ..})` as a base function and passing the relevant `props` in the test. 
image

That would also be extendable to other tests which we'll migrate, and would be easier to understand (from the perspective on how the current tests are written).

**Review by ratik21 — APPROVED** [review]
had a doubt over the test. but shouldn't hold up the merge!

File: src/components/admin-message/container.vitest.tsx (hunk 5 above)
Comment (colbr) [reply src/components/admin-message/container.vitest.tsx:33]: Can for sure create a reusable render function. I just didn't do it here as there are only 2 tests - for a larger test suite I'd definitely make a function

**Pull request merged**